src/uu/sleep/src/sleep.rs

```diff
@@ -61,14 +62,35 @@ pub fn uu_app() -> Command {
 
 fn sleep(args: &[&str]) -> UResult<()> {
     let mut arg_error = false;
+
+    use fundu::TimeUnit::*;
+    let parser = DurationParser::with_time_units(&[Second, Minute, Hour, Day]);
+
     let sleep_dur = args
         .iter()
-        .filter_map(|input| {
-            uucore::parse_time::from_str(input.trim()).ok().or_else(|| {
+        .filter_map(|input| match parser.parse(input.trim()) {
+            Ok(duration) => Some(duration),
+            Err(error) => {
                 arg_error = true;
-                show_error!("invalid time interval '{input}'");
+
+                let reason = match error {
+                    ParseError::Empty if input.is_empty() => "Input was empty".to_string(),
+                    ParseError::Empty => "Found only whitespace in input".to_string(),
+                    ParseError::Syntax(pos, description)
+                    | ParseError::TimeUnit(pos, description) => format!(
+                        "{description} at position {}",
+                        pos.checked_add(1).unwrap_or(pos)
```

Comment:
Nit: This is just a `saturating_add` isn't it?

Reply:
oh yes, I'll change it to `saturating_add`!

> Nice! I'll update the branch to get rid of the clippy warnings.

Thanks!

Reply:
I rebased onto main after this change, so the merge commit is gone again.